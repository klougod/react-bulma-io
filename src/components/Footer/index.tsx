import React from 'react'
import PropTypes from 'prop-types'

export const Footer = ({ className, children }: any) => (
  <footer className={`footer ${className}`}>{children}</footer>
)

Footer.defaultProps = {
  className: ''
}

Footer.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node
}

export default Footer
