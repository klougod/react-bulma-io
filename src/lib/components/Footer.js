import React from 'react'
import PropTypes from 'prop-types'

export const Footer = props => {

  const {
    className,
    children
  } = props

  return (
    <footer className={`footer ${className}`}>
      { children }
    </footer>
  )
}

Footer.defaultProps = {
  className: ''
}

Footer.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node.isRequired
}

export default Footer