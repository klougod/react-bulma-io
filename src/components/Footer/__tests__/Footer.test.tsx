import React from 'react'
import Enzyme from 'enzyme'

import Footer from '../'

describe('Footer component test', function () {
  it('should render without crash without props', function () {
    const footerMock = Enzyme.shallow(<Footer />)
    expect(footerMock).toMatchSnapshot()
  })

  it('should render without crash with props', function () {
    const footerMockProps = {
      className: 'footerCLass'
    }
    const footerMock = Enzyme.shallow(<Footer {...footerMockProps} />)
    expect(footerMock).toMatchSnapshot()
  })
})
