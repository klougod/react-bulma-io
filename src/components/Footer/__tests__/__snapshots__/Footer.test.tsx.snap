// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component test should render without crash with props 1`] = `
<footer
  className="footer footerCLass"
/>
`;

exports[`Footer component test should render without crash without props 1`] = `
<footer
  className="footer "
/>
`;
